src/opera/util/metadata_utils.py

```diff
@@ -185,17 +186,50 @@ def get_geographic_boundaries_from_mgrs_tile(mgrs_tile_name):
     for offset_x_multiplier in range(2):
         for offset_y_multiplier in range(2):
 
-            x = x_min + offset_x_multiplier * 109.8 * 1000
-            y = y_min + offset_y_multiplier * 109.8 * 1000
+            # We are using MGRS 100km x 100km tiles
+            # HLS tiles have 4.9 km of margin => width/length = 109.8 km
+            x = x_min - 4.9 * 1000 + offset_x_multiplier * 109.8 * 1000
+            y = y_min - 4.9 * 1000 + offset_y_multiplier * 109.8 * 1000
+
             lat, lon, z = transformation.TransformPoint(x, y, elevation)
 
+            # wrap longitude values within the range [-180, +180]
+            if lon < -180:
+                lon += 360
+            elif lon > 180:
+                lon -= 360
+
             if lat_min is None or lat_min > lat:
                 lat_min = lat
             if lat_max is None or lat_max < lat:
                 lat_max = lat
-            if lon_min is None or lon_min > lon:
+
+            # The computation of min and max longitude values may be affected
+            # by antimeridian crossing. Notice that: 179 degrees +
+            # 2 degrees = -179 degrees
+            #
+            # The condition `abs(lon_min - lon) < 180`` tests if both longitude
+            # values are both at the same side of the dateline (either left
+            # or right).
+            #
+            # The conditions `> 100` and `< 100` are used to test if the
+            # longitude point is on the left side of the antimeridian crossing
+            # (`> 100`) or at the right side (`< 100`)
+            #
+            # We also want to check if the point is at the west or east side of the tile.
+            # Points at the west, i.e, where offset_x_multiplier == 0 may update `lon_min`
+            if (offset_x_multiplier == 0 and
```

Comment:
I think the if-statements for `lon_min` and `lon_max` could be simplified.  Add `offset_x_multiplier ==0 and ...` on the previous if-statement.  Something like this:
```
if (offset_x_multiplier == 0) and (lon_min is None or lon_min > lon):
    lon_min = lon
if (offset_x_multiplier == 1) and (lon_max is None or lon_max < lon):
    lon_max = lon
``` 

That should pick the `lon_min` from the two boundary points on the "left-hand" side of the granule (viewing when north is up), and `lon_max` from the two boundary points on the "right-hand" side.